micro_http/src/common/mod.rs

```diff
@@ -28,6 +28,17 @@ pub enum RequestError {
     InvalidRequest,
 }
 
+/// Errors associated with a HTTP Connection.
+#[derive(Debug, PartialEq)]
+pub enum ConnectionError {
+    /// The request parsing has failed.
+    ParseError(RequestError),
+    /// Could not perform a stream operation successfully.
+    StreamError,
```

Comment:
StreamError should be StreamError(std::io::error) and whenever there is a stream error this should be propagated. Otherwise it is super hard to do debugging.

For this change you will need to manually implement PartialEq as std::io::Error does not derive PartialEq.